src/idl_gen_cpp.cpp

```diff
@@ -578,7 +578,8 @@ class CppGenerator : public BaseGenerator {
                                bool is_constructor) {
     auto &ptr_type = PtrType(field);
     if (ptr_type != "naked") {
-      return ptr_type + "<" + type + ">";
+      return (ptr_type != "cpp_ptr_type" ? ptr_type :
```

Comment:
it's confusing that this pointer type has the same they key. maybe call it `default_ptr_type` ? And maybe also mention this in `CppUsage.md` under `# Using different pointer types.` ?

Reply:
Done